sources/engine/Stride.Graphics/Direct3D/GraphicsOutput.Direct3D.cs

```diff
@@ -205,23 +204,32 @@ private void InitializeCurrentDisplayMode()
         /// <returns>A matched <see cref="DisplayMode"/> or null if nothing is found.</returns>
         private DisplayMode TryFindMatchingDisplayMode(Format format)
         {
-            var desktopBounds = outputDescription.DesktopBounds;
-
-            foreach (var supportedDisplayMode in SupportedDisplayModes)
+            ModeDescription closestDescription;
+            SharpDX.Direct3D11.Device deviceTemp = null;
+            try
             {
-                var width = desktopBounds.Right - desktopBounds.Left;
-                var height = desktopBounds.Bottom - desktopBounds.Top;
-
-                if (supportedDisplayMode.Width == width
-                    && supportedDisplayMode.Height == height
-                    && (Format)supportedDisplayMode.Format == format)
-                {
-                    // Stupid DXGI, there is no way to get the DXGI.Format, nor the refresh rate.
-                    return new DisplayMode((PixelFormat)format, width, height, supportedDisplayMode.RefreshRate);
-                }
+                // We don't care about FeatureLevel because we want to get missing data 
+                // about the current display/monitor mode and not the supported display mode for the specific graphics profile
+                deviceTemp = new SharpDX.Direct3D11.Device(adapter.NativeAdapter);
             }
+            catch (Exception) { }
```

Comment:
In which case would it throw ? Might be best to still log it, perhaps through a global logger for this class;
```cs
private static readonly Logger Log = GlobalLogger.GetLogger(typeof(GraphicsOutput).FullName);
```

Reply:
It's a good idea, to be honest I don't know exactly in which cases it might throw, I followed the steps taken from FindClosestMatchingDisplayMode in the same file.